### ISSUE TITLE
vulkan support on M1, and "The GLFW library is not initialized"

### ISSUE THREAD

**Issue by ElonKou** [taichi-dev/taichi#5848]
Thank you very much for your great work. I'm running with Taichi [https://github.com/chunleili/tiPBD](https://github.com/chunleili/tiPBD) It seems that there is a problem with the support of Apple M1 Vulkan.

**To Reproduce**
clone https://github.com/chunleili/tiPBD 

OS : apple M1 
[Taichi] version 1.1.3, llvm 10.0.0, commit 0fe89820, osx, python 3.9.7

the code can't run on taichi 1.0.4 /1.0.0/1.1.2/1.1.3(nightly)

**Log/Screenshots**
```bash
python3 tiPBD.py
[Taichi] version 1.1.3, llvm 10.0.0, commit 0fe89820, osx, python 3.9.7
[Taichi] Starting on arch=arm64
[I 08/23/22 17:14:50.634 41223] [vulkan_device_creator.cpp:pick_physical_device@372] Found Vulkan Device 0 (Apple M1)
[I 08/23/22 17:14:50.634 41223] [vulkan_device_creator.cpp:create_logical_device@440] Vulkan Device "Apple M1" supports Vulkan 0 version 1.1.211
[W 08/23/22 17:14:50.634 41223] [vulkan_device_creator.cpp:create_logical_device@488] Potential non-conformant Vulkan implementation, enabling VK_KHR_portability_subset
[W 08/23/22 17:14:50.634 41223] [vulkan_device_creator.cpp:create_logical_device@556] Taichi GPU GUI requires wide lines support
/opt/homebrew/lib/python3.9/site-packages/taichi/ui/ui.py:22: DeprecationWarning: `ti.ui.make_camera()` is deprecated, please use `ti.ui.Camera()` instead
  warnings.warn(
[W 08/23/22 17:14:50.869 41223] [vulkan_device.cpp:buffer@620] Overriding last binding
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: vkAllocateMemory(): External memory requires a dedicated VkBuffer or VkImage.
[E 08/23/22 17:14:50.869 41223] [vulkan_cpu_interop.cpp:memcpy_cpu_to_vulkan_via_staging@46] Not supported.


Traceback (most recent call last):
  File "/Users/elonkou/FILE/TOOLS/tiPBD/tiPBD.py", line 130, in <module>
    main()
  File "/Users/elonkou/FILE/TOOLS/tiPBD/tiPBD.py", line 126, in main
    canvas.scene(scene)
  File "/opt/homebrew/lib/python3.9/site-packages/taichi/ui/canvas.py", line 126, in scene
    self.canvas.scene(scene.scene)
RuntimeError: [vulkan_cpu_interop.cpp:memcpy_cpu_to_vulkan_via_staging@46] Not supported.
GLFW Error 65537: The GLFW library is not initialized
```

**Comment by ailzhang** [comment]
Thanks for reporting! This was caused by a wrong build configuration, sending a fix soon!

**Comment by ElonKou** [comment]
> Thanks for reporting! This was caused by a wrong build configuration, sending a fix soon!

Thank you!

**Comment by ailzhang** [comment]
Fixed via #5865 , @ElonKou if you install the latest nightly this bug should be gone ;)
